ui/components/app/assets/asset-list/network-filter/network-filter.tsx

```diff
@@ -0,0 +1,144 @@
+
+  // TODO: fetch balances across networks
+  // const multiNetworkAccountBalance = useMultichainAccountBalance()
+
+  const handleFilter = (chainFilters: Record<string, boolean>) => {
+    dispatch(setTokenNetworkFilter(chainFilters));
+
+    // TODO Add metrics
+    handleClose();
+  };
+
+  return (
+    <>
+      <SelectableListItem
+        isSelected={!Object.keys(tokenNetworkFilter).length}
+        onClick={() => handleFilter({})}
+      >
+        <Box
+          display={Display.Flex}
+          justifyContent={JustifyContent.spaceBetween}
+        >
+          <Box>
+            <Text
+              variant={TextVariant.bodyMdMedium}
+              color={TextColor.textDefault}
+            >
+              {t('allNetworks')}
+            </Text>
+            <Text
+              variant={TextVariant.bodyMdMedium}
+              color={TextColor.textDefault}
+            >
+              {/* TODO: Should query cross chain account balance */}
+              $1,000.00
+            </Text>
+          </Box>
+          <Box display={Display.Flex}>
+            {Object.values(allNetworks)
+              .slice(0, 5) // only show a max of 5 icons overlapping
+              .map((network, index) => {
+                const networkImageUrl =
+                  CHAIN_ID_TO_NETWORK_IMAGE_URL_MAP[
+                    network.chainId as keyof typeof CHAIN_ID_TO_NETWORK_IMAGE_URL_MAP
+                  ];
+                return (
+                  <AvatarNetwork
+                    key={network.chainId}
+                    name="All"
+                    src={networkImageUrl ?? undefined}
+                    // overlap the icons
+                    style={{
+                      marginLeft: index === 0 ? 0 : '-20px',
+                      zIndex: 5 - index,
+                    }}
+                  />
+                );
+              })}
+          </Box>
+        </Box>
+      </SelectableListItem>
+      <SelectableListItem
+        isSelected={tokenNetworkFilter[chainId]}
+        onClick={() => handleFilter({ [chainId]: true })}
+      >
+        <Box
+          display={Display.Flex}
+          justifyContent={JustifyContent.spaceBetween}
+        >
+          <Box>
+            <Text
+              variant={TextVariant.bodyMdMedium}
+              color={TextColor.textDefault}
+            >
+              {t('currentNetwork')}
+            </Text>
+            <UserPreferencedCurrencyDisplay
+              value={selectedAccountBalance}
+              type="PRIMARY"
+              ethNumberOfDecimals={4}
+              hideTitle
+              shouldCheckShowNativeToken
+              isAggregatedFiatOverviewBalance={
+                !showNativeTokenAsMainBalance && !isTestnet
+              }
+            />
+          </Box>
+          <AvatarNetwork
+            name="Current"
+            src={currentNetwork?.rpcPrefs?.imageUrl}
+          />
+        </Box>
+      </SelectableListItem>
```

Comment:
as future improvement , we need to implement a storybook + unit test for this.
can you add a todo for that ?